blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DatastaxMetadataIO.java

```diff
@@ -0,0 +1,162 @@
+
+        Timer.Context ctx = Instrumentation.getReadTimerContext( CassandraModel.CF_METRICS_METADATA_NAME );
+
+        Session session = DatastaxIO.getSession();
+
+        try {
+
+            List<String> query = new ArrayList<String>();
+
+            for( Locator l : locators) {
+                query.add( l.toString() );
+            }
+
+            Select.Where select = select()
+                    .all()
+                    .from( CassandraModel.CF_METRICS_METADATA_NAME )
+                    .where( in( KEY, query ) );
+
+            List<Row> results = session.execute( select ).all();
+
+            Table<Locator, String, String> metaTable = HashBasedTable.create();
+
+            for ( Row row : results ) {
+                if ( LOG.isDebugEnabled() ) {
+                    LOG.debug( "Read metrics_metadata: " +
+                            row.getString( KEY ) +
+                            row.getString( COLUMN1 ) +
+                            serDes.deserialize(  row.getBytes( VALUE ) ) );
+                }
+
+                metaTable.put( Locator.createLocatorFromDbKey( row.getString( KEY ) ), row.getString( COLUMN1 ), serDes.deserialize(  row.getBytes( VALUE ) ) );
+            }
+
+            return metaTable;
+        }
+        finally {
+            ctx.stop();
+        }
+    }
+
+    @Override
+    public void put( Locator locator, String key, String value ) throws IOException {
+
+        Timer.Context ctx = Instrumentation.getWriteTimerContext( CassandraModel.CF_METRICS_METADATA_NAME );
+
+        Session session = DatastaxIO.getSession();
+
+        try {
+            Insert insert = insertInto( CassandraModel.CF_METRICS_METADATA_NAME )
+                    .value( KEY, locator.toString() )
+                    .value( COLUMN1, key )
+                    .value( VALUE, serDes.serialize( value ) );
+            // TODO: This is required by the cassandra-maven-plugin 2.0.0-1, but not by cassandra 2.0.11, which we run.
+            // I believe its due to the bug https://issues.apache.org/jira/browse/CASSANDRA-6238
+            insert.setConsistencyLevel( ConsistencyLevel.ONE );
+
+            ResultSet result = session.execute( insert );
+            LOG.debug( "result.size=" + result.all().size() );
+        }
+        finally {
+            ctx.stop();
+        }
+    }
+
+    @Override
+    public void putAll( Table<Locator, String, String> meta ) throws IOException {
+
+        Timer.Context ctx = Instrumentation.getWriteTimerContext( CassandraModel.CF_METRICS_METADATA_NAME );
+
+        Session session = DatastaxIO.getSession();
+
+        // TODO:  We need to look into doing async calls on each row.  Batch supposedly is stupid in Cassandra
+        // when your keys aren't on the same partition
+        try {
+            PreparedStatement ps = session.prepare( "INSERT INTO " + CassandraModel.CF_METRICS_METADATA_NAME + " (key, column1, value) VALUES (?, ?, ?)" );
```

Comment:
How are you getting by without specifying a keyspace?  Do you have a default connection setup in your blueflood.conf or something?  I am getting  "No keyspace has been specified." error if I do it the same way you have done here for a different table.